src/main/perl/Path.pm

```diff
@@ -346,6 +346,54 @@ sub any_exists
     return $path && (-e $path || -l $path);
 }
 
+=item is_symlink
+
+Test if C<path> is a symlink.
+
+=cut
+
+sub is_symlink
+{
+    my ($self, $path) = @_;
+    return $path && -l $path;
+}
+
+=item make_symlink
+
+Create a symlink C<path> whose target is C<target>.
+
+Returns an error if C<path> already exists and is not a 
+symlink. If it exists and is a symlink, it is updated.
+
+Returns SUCCESS on sucess, undef otherwise.
+
+=cut
+
+sub make_symlink
+{
+    my ($self, $target, $path) = @_;
+    my $status = SUCCESS;     # Assume success
+    unless ($path) {
+        $self->error("create_symlink(): 'path' argument missing");
+    }
+    unless ($target) {
+        $self->error("create_symlink(): 'target' argument missing");
+    }
+
+    if ( $self->is_symlink($path) ) {
+        $self->debug(1, "Removing symlink $path");
+        unlink ($path);
+    } elsif ( $self->any_exists($path) ) {
+        $self->error("$path already exists and is not a symlink");
+        return;
+    }
+
+    $status = symlink $target, $path;
+    $status = SUCCESS if $status;
```

Comment:
success should be eturned when nothing has changed (eg link already existed and pointed to same target).
return CHANGED if a link was actually created